KA-Flocking/Assets/Scripts/Castle Spawning/SpawnQueue.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.SceneManagement;
+
+public class SpawnQueue : MonoBehaviour
+{
+    public Transform spawnPoint;
+    public GameObject defaultItem;
+    [System.NonSerialized]
+    public List<GameObject> currentSpawnedItems = new List<GameObject>();
+    public RectTransform content;
+    private Button addButton;
+    [System.NonSerialized]
+    public Castle castle = null;
+    public ErrorChat errorChat;
+    public Flock flock;
+    public Text money;
+    // Timer is not applicable since we cannot see the queue when the game is running
+    // public Image timer;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        if (SceneManager.GetSceneByName("PlayerOneSetupScene").isLoaded) { 
+            flock = GameObject.Find("Team 1 Flock").GetComponent<Flock>();
+        } else if (SceneManager.GetSceneByName("PlayerTwoSetupScene").isLoaded) {
+            flock = GameObject.Find("Team 2 Flock").GetComponent<Flock>();	
+        }
+        money.text = "Money: " + flock.moneyAmount.ToString(); 
+        addButton = GetComponentInChildren<Button>(spawnPoint);
+        ToggleGroup troopToggles = GameObject.Find("TroopToggles").GetComponent<ToggleGroup>();
+        Slider troopSlider = GameObject.Find("troopAmountSlider").GetComponent<Slider>();
+        addButton.onClick.AddListener(delegate () {this.addButtonClicked(troopToggles, troopSlider);});
+        // timer = addButton.transform.GetChild(0).GetComponent<Image>();
+    }
+
+    /*public void ResetTimer(float spawnTime, Castle castle) {
+        if (castle == this.castle) {
+            timer.fillAmount = 0;
+            StartCoroutine(UpdateTimer(spawnTime));
+        }
+    }
+    private IEnumerator UpdateTimer(float spawnTime) {
+        while (timer.fillAmount < 1) {
+            timer.fillAmount += Time.deltaTime/spawnTime;
+            yield return new WaitForEndOfFrame();
+        }
+    }*/
```

Comment:
This code existed in order to make a "How long until the next spawn", but that seems irrelvant since u can't see the spawnqueue during the flock scene and the time is paused during the setup scene. You can see what units are in queue if you select the castle, but ofc you can't atm do that during the flock scene

Seems like the main point is that it seems irrelevant to use the spawn queue and that it's not visible during the flock scene. I'll put this as a pointer on the next meeting